### PR TITLE
Button refactor

### PR DESCRIPTION
**Pull request by benjag**
- [x] Script setup
- [x] Tests pass
- [x] BuildClass removed
- [x] Remove elevated? Yes, design sign off

### CODE REVIEW

File: src/components/button/examples/demo/CTA.vue [hunk 1]
```diff
@@ -14,7 +14,6 @@
         :key="index2"
         :size="button.size"
         :full-width="button.fullWidth"
-        :disabled="button.disabled"
```
Comment (benjag) [inline src/components/button/examples/demo/CTA.vue:17]: This is not a valid attribute value (nor is it needed for this example). I think the JSX compilation must have ignored it

File: src/components/button/styles/CdrButton.module.scss [hunk 2]
```diff
@@ -119,7 +119,7 @@
       @include cdr-button-large-mixin;
     }
 
-    &--fullWidth\@xs {
+    &--full-width\@xs {
```
Comment (benjag) [inline src/components/button/styles/CdrButton.module.scss:122]: For some reason, this was camelCased here where every other style modifier is kebab-cased. Changing for consistency plus the JS generates a kebab'd class ... that was a fun bug to chase down.

File: src/components/button/styles/vars/CdrButton.vars.scss [hunk 3]
```diff
@@ -389,7 +380,7 @@
 }
 
 @mixin cdr-button-large-mixin() {
-  @include cdr-button-base-text-mixin-large;
+  @include cdr-button-base-text-mixin-medium;
```
Comment (benjag) [inline src/components/button/styles/vars/CdrButton.vars.scss:383]: This is just matching a change we made to prod

File: src/components/button/CdrButton.vue [hunk 4]
```diff
@@ -21,105 +97,5 @@
...
-<!-- NOTE: REMOVED icon slot. use icon-left -->
-<script>
-import { defineComponent, useCssModule, computed } from 'vue';
-
-import mapClasses from '../../utils/mapClasses';
-import { buildClass, buildBooleanClass } from '../../utils/buildClass';
-import propValidator from '../../utils/propValidator';
-
-export default defineComponent({
-  name: 'CdrButton',
```
Comment (KenjiCrosland) [inline src/components/button/CdrButton.vue:33]: Interesting. Now that we're not providing a name property are we getting it from the filename?

File: src/components/button/CdrButton.vue [hunk 5]
```diff
@@ -1,3 +1,79 @@
+<script setup>
```
Comment (KenjiCrosland) [inline src/components/button/CdrButton.vue:1]: Nice. The `<script setup>` syntax is a lot easier to parse and remember what goes where. It's definitely an improvement over what we're replacing.

**Review by KenjiCrosland — APPROVED** [review]

File: src/components/button/CdrButton.vue (hunk 4 above)
Comment (benjag) [reply src/components/button/CdrButton.vue:33]: No, `<script setup>` doesn't allow the name prop. If for some reason we end up needing it, you can declare it within a normal script block.

**Pull request merged**